features/step_definitions/api/plan_features_steps.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+When('I {enable} the plan feature {string}') do |enable, feature_name|
+  within(:xpath, "//tr[td[text() = '#{feature_name}' and @class='title']]") do
+    if enable
+      find('i.excluded').click
+    else
+      find('i.included').click
+    end
+  end
+end
```

Comment:
This step is unnecessary and prone tu error (enabling a feature twice will raise an error). Moreover, if the step says "I enable feature X" then it is redundant to check "feature is enabled" right after.

It should be possible to use `click feature_x within features_selector`.

Reply:
> This step is unnecessary and prone tu error (enabling a feature twice will raise an error).

Well, I think that's fine. The user's action is specific - either enable or disable. If the step is "enable the feature", but the "enable" button is not visible, that means that the background was not set up properly, and I would expect this step to fail.

> Moreover, if the step says "I enable feature X" then it is redundant to check "feature is enabled" right after.

I don't think it's redundant. `I {enable} the plan feature {string}` triggers a server action, which may work as expected, or may not. The `I see the plan feature {string} is {enabled}` verifies that after executing the previous step achieved the desired result.

> It should be possible to use `click feature_x within features_selector`.

But what element do you suggest to click on? The clickable item that is tested in this scenario is this cross/check (toggable) icon in the **Enabled?** column:
![image](https://github.com/3scale/porta/assets/1270328/86e05844-e1a9-4552-94f2-1a79bcbd405f)

I don't think the action should be generic. For example, if the feature is already disabled, I will actually only see the "enabled" (checkbox) icon. Clicking on "icon in the Enabled? column" IMO is confusing.

Reply:
You could add the assertion `assert_selector('i.included[title="Feature is enabled for this plan"]', visible: true)` inside the `I {enable} the plan feature {string}` step. I saw that in many steps before ([example](https://github.com/3scale/porta/blob/36cf0b2b1a633d358ddb7d4ca57b6bac3323a282/features/step_definitions/backend_api_steps.rb#L64-L68), [example](https://github.com/3scale/porta/blob/a788c5bb5aa02890e98ce90b6ea468a3bffe202d/features/step_definitions/session_steps.rb#L48-L55)), but not sure that's a good practice or the expected way to do it.

You should check for the flash message anyway.

Reply:
To me this practice seems a bit weird, I prefer to separate the action and checking the outcome. But I am not a Cucumber expert in any way :grimacing: